controller/konnect/reconciler_consumerref.go

```diff
@@ -103,23 +103,9 @@ func handleKongConsumerRef[T constraints.SupportedKonnectEntityType, TEnt constr
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	if ok, err := controllerutil.HasOwnerReference(consumer.OwnerReferences, ent, cl.Scheme()); err != nil {
-		ctrllog.FromContext(ctx).Info("failed to check if KongConsumer has owner reference", "error", err)
-	} else if ok {
-		// Make sure there is no ownership between the consumer and the entity by removing
-		// a reference if found. Such references may have been added by older versions
-		// of KGO (< 1.5).
-		// TODO: remove this after a couple of minor versions.
-		old := ent.DeepCopyObject().(TEnt)
-		if err := controllerutil.RemoveOwnerReference(&consumer, ent, cl.Scheme()); err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to delete owner reference: %w", err)
-		}
-		if err := cl.Patch(ctx, ent, client.MergeFrom(old)); err != nil {
-			if k8serrors.IsConflict(err) {
-				return ctrl.Result{Requeue: true}, nil
-			}
-			return ctrl.Result{}, fmt.Errorf("failed to remove owner reference: %w", err)
-		}
+	res, err := RemoveOwnerRefIfSet(ctx, cl, ent, &consumer)
+	if err != nil || !res.IsZero() {
+		return ctrl.Result{}, err
```

Comment:
```suggestion
		return res, err
```
since `res` is not zero, it means it's not `ctrl.Result{}`, thus it contains e.g. requeue after. Or am I missing something?

Reply:
Thanks for catching this. Indeed this was supposed to return `res`. PTAL